src/tests/unit/electron/device-connect-view/components/device-connect-port-entry.test.tsx

```diff
@@ -14,4 +20,51 @@ describe('DeviceConnectPortEntryTest', () => {
 
         expect(rendered.getElement()).toMatchSnapshot();
     });
+
+    test('validate click fetch succeeds', () => {
+        const fetch: FetchScanResults = _ => Promise.resolve({ deviceName: 'dev', appIdentifier: 'app' } as ScanResults);
+        ValidatePortValidateClick(fetch, true, 'dev - app');
+    });
+
+    test('validate click fetch fails', () => {
+        const fetch: FetchScanResults = _ => Promise.reject({} as ScanResults);
+        ValidatePortValidateClick(fetch, false, undefined);
+    });
+
+    const ValidatePortValidateClick = (fetch: FetchScanResults, expectedSuccess: boolean, expectedDevice: string) => {
+        const testPortNumber = 111;
+        const eventStub = new EventStubFactory().createMouseClickEvent() as React.MouseEvent<Button>;
+        const fetchScanResultsMock = Mock.ofInstance(fetch);
```

Comment:
Can you move the creation of the mocks and their setting up in `beforeEach` or `beforeAll` blocks? How are you resetting the mocks for different test cases?

Reply:
The mocks are created and set up each time this function is called, which happens during the "validate click fetch succeeds" and "validate click fetch fails" tests. I didn't want to pull their creation/setup out to `beforeEach` because they aren't necessary for all of the tests and the setup of some of the mocks varies by test. Considering that, you say it's still worth pulling out the creation (not setup) of the mocks to `beforeAll`?

Reply:
Refactored to use `beforeAll` + split up the large function